aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1919,20 +1919,29 @@ static void apply_lu_batched_magma(const Tensor& input, const Tensor& pivots, co
 #endif
 }
 
-static void lu_magma(const Tensor& input, const Tensor& pivots, const Tensor& infos, bool compute_pivots) {
-  // TODO: compare performance and use the best performing option based on input's sizes
-  if (input.dim() == 2) {
+static void apply_lu(const Tensor& input, const Tensor& pivots, const Tensor& infos, bool compute_pivots) {
+  int64_t batch_size = batchCount(input);
+#ifdef USE_CUSOLVER
+  // Use a heuristic to determine that cusolver is faster than MAGMA for the following sizes.
```

Comment:
The heuristic data you've provided is really excellent, @v0dro. @xwang233, is there some way we can track the request for cuSOLVER to improve its performance where MAGMA is faster today? Maybe on our existing math library request issue?

Reply:
Sure. I can write a comment on that issue later.